userspace/libscap/engine/modern_bpf/scap_modern_bpf.c

```diff
@@ -143,13 +143,50 @@ static int32_t scap_modern_bpf__next(struct scap_engine_handle engine, OUT scap_
 	return SCAP_SUCCESS;
 }
 
+static int32_t scap_modern_bpf_start_dropping_mode(struct scap_engine_handle engine, uint32_t sampling_ratio)
+{
+	struct modern_bpf_engine *handle = engine.m_handle;
+	switch(sampling_ratio)
```

Comment:
What about moving this check in scap, instead of replicating it inside every driver that supports `SCAP_SAMPLING_RATIO`?

Reply:
Sure we can do that!

Reply:
Thanks !